indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSource.java

```diff
@@ -123,13 +129,23 @@ CloseableIterator<InputEntity> createEntityIterator()
       private Iterator<OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType, RecordType>> recordIterator;
       private Iterator<? extends ByteEntity> bytesIterator;
       private volatile boolean closed;
+      private long createTime = System.currentTimeMillis();
+      private Long terminationTime = samplerConfigTimeoutMs != null ? createTime + samplerConfigTimeoutMs : null;
 
       private void waitNextIteratorIfNecessary()
       {
         while (!closed && (bytesIterator == null || !bytesIterator.hasNext())) {
           while (!closed && (recordIterator == null || !recordIterator.hasNext())) {
+            if (terminationTime != null) {
+              if (System.currentTimeMillis() > terminationTime) {
+                LOG.info("Configured sampler timeout has been exceeded, returning without a bytesIterator.");
+                bytesIterator = null;
+                return;
+              }
+            }
```

Comment:
Would it make sense to combine these?

```suggestion
            if (terminationTime != null && System.currentTimeMillis() > terminationTime) {
              LOG.info("Configured sampler timeout has been exceeded, returning without a bytesIterator.");
              bytesIterator = null;
              return;
            }
```